packages/main/src/index.ts

```diff
@@ -37,6 +40,53 @@ function createWindow() {
   })
 }
 
+function handleContentDownloads() {
+  protocol.registerStreamProtocol("body", async (request, callback) => {
+    try {
+      const parsed = parseBodyUri(request.url)
+      if (!parsed) {
+        throw new Error(
+          `Unable to parse messageId and partId from URI: ${request.url}`
+        )
+      }
+      const messageId = parsed.messageId
+      const partId = parsed.partId
```

Comment:
If you want you combine this sort of assignment into one statement using destructuring:

```ts
const { messageId, partId } = parsed
```